### PR TITLE
MIME4j-268 MIME4J-269 Convenience helpers...

### PR DESCRIPTION
**Pull request by chibenwa**
- Expose a LENIENT default config
 - Provide a helper method for Message to byte[] conversion

### CODE REVIEW

File: core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java [hunk 1]
```diff
@@ -26,6 +26,12 @@
  */
 public final class MimeConfig {
 
+    private static final MimeConfig LENIENT = MimeConfig.custom()
```
Comment (aduprat) [inline core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: Why private?

Comment (chibenwa) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: Oups yes should of course be public....

**Review by rouazana — APPROVED** [review]

**Review by aduprat — APPROVED** [review]

File: core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java (hunk 1 above)
Comment (mbaechler) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: It's not only lenient (and btw you don't even call `setStringParsing(false)` ) but dangerous : max values are here to protect use from malicious input.
What happens with a mail crafted to overflow the parser ? Can I trigger JVM OutOfMemoryException with a single mail ?

Comment (chibenwa) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: I guess so... 

Then maybe I should drop MIME4J-269, and introduce a ticket for defining in details constants to be used safely, while not failing on legitimate email, that can be used by James.

What do you think?

Comment (mbaechler) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: Not sure to understand

Comment (chibenwa) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: I mean, as you mention, dropping all limits to have no input limits is unsafe.

We'd better use a mime-config carefully defining a "safe enough/lenient enough" limit instead that we should be using in James.

Comment (mbaechler) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: agree

Comment (chibenwa) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: I propose 100 M as a safe upper bound for **maximum content length** and keep the other ones unlimited.

This should be enough to not overwhelm the JVM too much while still being very permissive.

We can call such a constant "PERMISSIVE".

WDYT?

Comment (mbaechler) [reply core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java]: why not, needs some testing to check what the actual memory consumption is.

**Review by aduprat — APPROVED** [review]

File: dom/src/test/java/org/apache/james/mime4j/dom/LargeMessageParsingTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        outputStream.write("\r\n".getBytes());
+        // 38 * 1.600.000 = ~ 58 Mo of body
+        for (int i = 0; i < 1600000; i++) {
+            outputStream.write(String.format("abcdeghijklmnopqrstuvwxyz0123456789\r\n", i, i).getBytes());
+        }
+
+        DefaultMessageBuilder messageBuilder = new DefaultMessageBuilder();
+        messageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE);
+        messageBuilder.parseMessage(new ByteArrayInputStream(outputStream.toByteArray()));
+    }
```
Comment (rouazana) [inline dom/src/test/java/org/apache/james/mime4j/dom/LargeMessageParsingTest.java:47]: can you add some tests with (very) long lines please?

**Review by rouazana — APPROVED** [review]

**Review by aduprat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**